streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -261,12 +260,18 @@ public void onPartitionsAssigned(final Collection<TopicPartition> assignment) {
                 taskManager.suspendedActiveTaskIds(),
                 taskManager.suspendedStandbyTaskIds());
 
+            if (streamThread.assignmentErrorCode.get() != 0) {
```

Comment:
This condition seems not right: we should not shutdown if the error was `VERSION_PROBING`, right? I.e we should firstly check if the error was `NONE`, and if not, switch branch on the actual error code to handle them accordingly.